### DISCUSSION TITLE
Forms : Single choice dropdown for hours and minutes

### DISCUSSION THREAD

**Discussion by edgarogh** [symfony/symfony#47392] [Q&A]
Hello,

Is there an easy way to have a single choice component for hours and minutes instead of two. In my specific case, the only possible values are 15 minutes increments during business hours, and I believe it would be a lot faster for the user to have a single choice/spinner/dropdown and just scroll once to the desired time. I already know that I can use `hours` and `minutes` to restrict their possible values, it's the choice part that I'm looking a solution for.

<details>
<summary>Example values</summary>

```
08:00
08:15
08:30
08:45
09:00
09:15
09:30
09:45
10:00
...
18h00
```

</details>

If that's not integrated into Symfony, I know how to implement a sub-form and can work my way on my own, especially since I can base my code on Symfony's.

Thank you

**Comment by MatTheCat** [comment]
Hi!

AFAIK you cannot achieve such UI using Symfony’s `TimeType`.

That being said it seems your use case only require  a `ChoiceType` field?

**Reply by edgarogh** [reply]
I need a choice type, but actually this is part of a datetime picker bound to a class member. So I also need to handle the date picker part and do the necessary conversion to end up with a `DateTime` or its immutable variant. And I'd like it to be reusable.

That's completely fine, I'll do the job myself. I was really just wondering if there was a simple boolean flag for this.

**Reply by MatTheCat** [reply]
Not that I know of :grimacing: 

Given we cannot even use `time` inputs `step` attribute it could be an interesting feature request though :thinking:
